### PR TITLE
Input validation

### PR DESCRIPTION
**Pull request by sspangenberg**
Closes: #77

### CODE REVIEW

File: app/meetings/components/schedules/AddScheduleModal.tsx [hunk 1]
```diff
@@ -39,6 +41,36 @@ const AddSchedule = (props: AddScheduleProps) => {
     setName(e.currentTarget.value)
   }
 
+  const checkSchedules = (schedules) => {
+    let result = true
+    for (const key in schedules) {
+      if (!schedule[key].blocked) {
+        if (!checkTime(schedule[key].start) || !checkTime(schedule[key].end)) {
+          result = false
```
[the remark this answers is not in the document]
Comment (sspangenberg) [reply app/meetings/components/schedules/AddScheduleModal.tsx]: Done

**Comment by sspangenberg** [comment]
@phlprcks please check whether the validation of schedule times work as expected

File: app/auth/validations.ts [hunk 2]
```diff
@@ -41,3 +47,42 @@ export const UpdateUserInput = z.object({
     .max(100, { message: messages.long_password }),
 })
 export type UpdateUserInputType = z.infer<typeof UpdateUserInput>
+
+const Schedule = z.object({
+  blocked: z.boolean(),
+  start: z.string(),
+  end: z.string(),
```
Comment (phlprcks) [inline app/auth/validations.ts:54]: Looks great, but one check is missing: I can enter 23:00 - 17:00. The first time should be before the 2nd one, right? 🤔 

Also, as discussed in (#153 ) that we should adapt to user preferences in time format, that's why I created #167 . We would need a different validation logic once that is implemented, but that can be part of #167 in my opinion.

File: app/meetings/components/schedules/AddScheduleModal.tsx [hunk 3]
```diff
@@ -61,6 +92,18 @@ const AddSchedule = (props: AddScheduleProps) => {
       },
     }
 
+    const parseResult = ScheduleInput.refine((data) => checkSchedules(data.schedule), {
+      message: "Please check the entered times",
```
Comment (phlprcks) [inline app/meetings/components/schedules/AddScheduleModal.tsx:96]:
```suggestion
      message: "Please check the entered times. Expected is a format of hour:minutes, e.g. 09:30",
```

File: app/meetings/components/schedules/AddScheduleModal.tsx [hunk 4]
```diff
@@ -39,6 +41,35 @@ const AddSchedule = (props: AddScheduleProps) => {
...
+  const checkTime = (time) => {
+    const values = time.split(":")
+    if (values.length === 2) {
+      if (values[0].trim().length === 2 && values[1].trim().length === 2) {
+        const hour = parseInt(values[0])
+        const minute = parseInt(values[1])
+        if (!Number.isNaN(hour) && !Number.isNaN(minute)) {
+          if (hour >= 0 && hour <= 23) {
+            if (minute >= 0 && minute <= 59) {
+              return true
```
Comment (phlprcks) [inline app/meetings/components/schedules/AddScheduleModal.tsx:64]: see above

**Review by phlprcks — CHANGES_REQUESTED** [review]
how about validating the input when users book a meeting from the schedule link? I have some provisional logic for that in #165 but it would make sense to use zod for this validation as well 👍

File: app/auth/validations.ts (hunk 2 above)
Comment (sspangenberg) [reply app/auth/validations.ts:54]: I agree that all time format related tasks should be part of #167. The time comparison is implemented in the latest commit.

**Comment by sspangenberg** [comment]
> 
> 
> how about validating the input when users book a meeting from the schedule link? I have some provisional logic for that in #165 but it would make sense to use zod for this validation as well 👍

 If you have aready worked on it I would implement it on your branch due to simplicity

**Review by phlprcks — APPROVED** [review]

**Pull request merged**